mods/ts/weapons/energyweapons.yaml

```diff
@@ -39,17 +75,42 @@ MechRailgun:
 		DamageTypes: Prone100Percent, TriggerProne, FireDeath
 
 SonicZap:
-	ReloadDelay: 120
+	ReloadDelay: 180
 	Range: 6c0
 	Charges: yes
 	Report: SONIC4.AUD
-	Projectile: LaserZap
-		Color: 200,0,255,255
-		BeamWidth: 12
-		BeamDuration: 50
+	Projectile: LineImpactProjectile
+		Speed: 0c64
+		TicksBetweenImpacts: 5 # Roughly 18 impacts.
+		BeyondTargetRange: 0c256
+		MaxTargetRange: 50c0
+		FalloffMaxReduction: 0
+		FalloffStartDistance: 49c0
+		FalloffEndDistance: 50c0
+		BeamWidth: 0c768
+		Blockable: true
+		BeamRenderWidth: 768
+		BeamFireDuration: 90
+		Color: 200,0,40,120
```

Comment:
You should be able to revert these color changes now that the translucency works properly (although I haven't tested this since you fixed that).